MapboxNavigation/NavigationMapView.swift

```diff
@@ -1016,7 +1002,9 @@ public protocol NavigationMapViewDelegate: class  {
     
     @objc optional func navigationMapView(_ mapView: NavigationMapView, routeCasingStyleLayerWithIdentifier identifier: String, source: MGLSource) -> MGLStyleLayer?
     
-    @objc optional func navigationMapView(_ mapView: NavigationMapView, didTap route: Route)
+    @objc optional func navigationMapView(_ mapView: NavigationMapView, didSelectRoute route: Route)
+    
+    @objc optional func navigationMapView(_ mapView: NavigationMapView, didSelectWaypoint waypoint: Waypoint)
```

Comment:
The signature should be `navigationMapView(_:didSelect:)` in Swift but `-navigationMapView:didSelectWaypoint:` in Objective-C. You can customize the Objective-C signature using `@objc(…)`.

Reply:
Good catch. Done.